### PR TITLE
Upload code coverage of unit tests to codecov

### LINKED ISSUE #6289: Track unit tests code coverage
Create a new workflow for tracking unit test code coverage, using the [codecov-action](https://github.com/codecov/codecov-action).

### PR DESCRIPTION
**Pull request by akshitgrover**
Code coverage is not being tracked for unittests, Tracking code coverage is important as it give insights on where to test and what to test

This PR updates unit tests workflow to create coverage reports and upload the same to codecov using codecov github action

Fixes #6289

Signed-off-by: Akshit Grover <akshit.grover2016@gmail.com>

### CODE REVIEW

**Comment by akshitgrover** [comment]
@alpeb 
Please review if this solves the code coverage tracking issue #6289

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -23,7 +23,12 @@ jobs:
       run: |
         go get gotest.tools/gotestsum@v0.4.2
         # TODO: validate bin/protoc-go.sh does not dirty the repo
-        gotestsum -- -cover -race -v -mod=readonly ./...
+        gotestsum -- -cover -coverprofile=coverage.out -race -v -mod=readonly ./...
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
```
Comment (alpeb) [inline .github/workflows/unit_tests.yml:28]: It's recommended to refer directly to the sha vs the tag (also please change in the other reference below):
```suggestion
      uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192
```

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -23,7 +23,12 @@ jobs:
...
+      with:
+        files: ./coverage.out
+        flags: unittests,golang
```
Comment (alpeb) [inline .github/workflows/unit_tests.yml:31]: The report shows:
```

Flag | Coverage Δ | &nbsp;
-- | -- | --
golang | 42.44% &lt;0.00%&gt; (?) | &nbsp;
javascript | 66.69% &lt;0.00%&gt; (?) | &nbsp;
unittests | 44.37% &lt;0.00%&gt; (?)
```
These are all unit tests, so I think the `unittests` flag is redundant.

**Review by alpeb — COMMENTED** [review]
Thanks @akshitgrover
I've triggered the tests, so you can check the results in the PR :wink:

**Comment by akshitgrover** [comment]
@alpeb 
I have updated the workflow to use SHA

I thought of adding the 'unittests' flag as it will give the overall coverage of the project whilst 'golang' and 'javascript' projects will point to coverage of .go and .js files respectively.
Removed the 'unittests' flag for now

**Review by alpeb — APPROVED** [review]
LGTM :+1: :slightly_smiling_face:

**Comment by akshitgrover** [comment]
Thank you @alpeb 
Who else should I ask for the review?
As it requires 2 people to review before I can merge

**Review by kleimkuhler — COMMENTED** [review]
@akshitgrover thanks for adding this! Could you actually add back the `unittests` flag to the golang and javascript tests? It'll be nice to keep that as a general flag so that `unittests` represents the total coverage of all unit tests.

**Comment by akshitgrover** [comment]
@kleimkuhler 
I have added back 'unittests' flag

**Review by kleimkuhler — APPROVED** [review]
Thanks @akshitgrover 🙂

**Pull request merged**